shared/src/main/java/com/couchbase/lite/Result.java

```diff
@@ -107,146 +107,164 @@ public int count() {
 
     @Override
     public Object getObject(int index) {
-        FLValue value = flValueAtIndex(index);
-        return value.toObject(null, null);
+        return fleeceValueToObject(index);
     }
 
     @Override
     public String getString(int index) {
-        return null;
+        return (String) fleeceValueToObject(index);
     }
 
     @Override
     public Number getNumber(int index) {
-        return null;
+        return (Number) fleeceValueToObject(index);
     }
 
     @Override
     public int getInt(int index) {
-        return 0;
+        return (int) fleeceValue(index).asInt();
     }
 
     @Override
     public long getLong(int index) {
-        return 0;
+        return fleeceValue(index).asInt();
     }
 
     @Override
     public float getFloat(int index) {
-        return 0;
+        return fleeceValue(index).asFloat();
     }
 
     @Override
     public double getDouble(int index) {
-        return 0;
+        return fleeceValue(index).asDouble();
     }
 
     @Override
     public boolean getBoolean(int index) {
-        return false;
+        return fleeceValue(index).asBool();
     }
 
     @Override
     public Blob getBlob(int index) {
-        return null;
+        return (Blob) fleeceValueToObject(index);
     }
 
     @Override
     public Date getDate(int index) {
-        return null;
+        return DateUtils.fromJson(getString(index));
     }
 
     @Override
     public ReadOnlyArrayInterface getArray(int index) {
```

Comment:
It should return ReadOnlyArray not ReadOnlyArrayInterface. Also IIRC, ReadOnlyArrayInterface is internal for Java.